ipatests/test_ipalib/test_rpc.py

```diff
@@ -341,3 +343,52 @@ def test_help_many_params(self):
                 "command 'system.methodHelp' takes at most 1 argument")
         else:
             raise AssertionError('did not raise')
+
+
+class test_rpcclient_context(PluginTester):
+    """
+    Test the context in `ipalib.rpc.rpcclient` plugin.
+    """
+    def setup(self):
+        try:
+            api.Backend.rpcclient.connect(ca_certfile='foo')
+        except (errors.NetworkError, IOError):
+            raise nose.SkipTest('%r: Server not available: %r' %
+                                (__name__, api.env.xmlrpc_uri))
+
+    def teardown(self):
+        if api.Backend.rpcclient.isconnected():
+            api.Backend.rpcclient.disconnect()
+
+    def test_context_cafile(self):
+        """
+        Test that ca_certfile is set in `ipalib.rpc.rpcclient.connect`
+        """
+        ca_certfile = getattr(context, 'ca_certfile', None)
+        assert_equal(ca_certfile, 'foo')
+
+    def test_context_principal(self):
+        """
+        Test that principal is set in `ipalib.rpc.rpcclient.connect`
+        """
+        principal = getattr(context, 'principal', None)
+        assert_equal(principal, 'admin@%s' % api.env.realm)
+
+    def test_context_request_url(self):
+        """
+        Test that request_url is set in `ipalib.rpc.rpcclient.connect`
+        """
+        request_url = getattr(context, 'request_url', None)
+        assert_equal(request_url, 'https://%s/ipa/session/json' % api.env.host)
+
+    def test_context_session_cookie(self):
+        """
+        Test that session_cookie is set in `ipalib.rpc.rpcclient.connect`
+        """
+        fuzzy_cookie = Fuzzy('^ipa_session=MagBearerToken=[A-Za-z0-9+\/+]=+;$')
```

Comment:
Your regexp is wrong. It should be `Fuzzy('^ipa_session=MagBearerToken=[A-Za-z0-9+/]+=+;$')` (you misplaced the positive iteration)

Reply:
Bah, that's what I guess for rushing. I also made trailing = to be *.

Reply:
Ah, right, I did not realize that, good catch.